subprojects/docs/src/docs/userguide/authoring-builds/software-product/composite_builds.adoc

```diff
@@ -101,13 +101,12 @@ However, this does not support all use cases and including plugin builds like th
 === Restrictions on included builds
 
 Most builds can be included into a composite, including other composite builds.
-However there are some restrictions.
+However, there are some restrictions.
 
 Every included build:
 
-* must not have a `rootProject.name` the same as another included build.
-* must not have a `rootProject.name` the same as a top-level project of the composite build.
-* must not have a `rootProject.name` the same as the composite build `rootProject.name`.
+* must not have a `rootProject.name` that is the same as a top-level project name of the composite build.
```

Comment:
I'm not sure any of these restrictions apply any more, thb.

Reply:
The first of the two remaining requirements listed here, I think, still applies:
> must not have a `rootProject.name` that is the same as a top-level project name of the composite build.

I presume what is meant here by "top-level project" is the first level of subprojects. This seems to be validated by an existing integration test:
https://github.com/gradle/gradle/blob/54825879d3213a77b317e2464c997b9798bd3fe4/subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/IncludedBuildValidationIntegrationTest.groovy#L87-L99

Similarly, the second remaining requirement is also tested.
> must not have a `rootProject.name` that is the same as the composite build `rootProject.name`.

https://github.com/gradle/gradle/blob/54825879d3213a77b317e2464c997b9798bd3fe4/subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/IncludedBuildValidationIntegrationTest.groovy#L101-L110

Is this right, or am I misinterpreting something?